### PR TITLE
Pre commit

### LINKED ISSUE #29: flake8-plugins
There are always more linting plugins one could use. Here are some I've found helpful:

- `flake8-comprehensions`: checks your list comprehensions for common mistakes.
- `flake8-black`: duh
- `flake8-builtins`: checks if you're naming variables overwriting builtins
- `flake8-eradicate`: checks for old commented code that you should just delete
- `pep8-naming`: checks naming of variables for pep8-compliance
- `flake8-isort`: checks sorting of imports
- `flake8-rst-docstrings`: checks docstrings for RST syntax (very helpful!)
- `flake8-copyright`: checks copyright
- `flake8-docstrings`: checks for common docstring conventions (like a one-line imperative summary)

### PR DESCRIPTION
**Pull request by steven-murray**
Adds pre-commit, and also a whole lot of automatic flake8 goodness.

Fixes #40 
Fixes #29 

Since this changes a lot of files, it might be a bit tricky to merge with other PRs on the loose. 

So far, I've added 

```
- flake8-docstrings
- flake8-rst-docstrings
- flake8-comprehensions
- flake8-black
- flake8-builtins
- flake8-copyright
```

and made changes in the code to make these pass.

I think we should also enable `pep8-naming` and `flake8-eradicate`, but I want to hear others' opinions first. Adding pep8-naming gives the following errors:

```
pyradiosky/skymodel.py:200:10: N806 variable 'R_screwy' in function should be lowercase
pyradiosky/skymodel.py:201:10: N806 variable 'R_really_orthogonal' in function should be lowercase
pyradiosky/skymodel.py:204:10: N806 variable 'R_really_orthogonal' in function should be lowercase
pyradiosky/skymodel.py:235:10: N806 variable 'R_avg' in function should be lowercase
pyradiosky/skymodel.py:237:10: N806 variable 'R_exact' in function should be lowercase
pyradiosky/skymodel.py:242:14: N806 variable 'R_perturb' in function should be lowercase
pyradiosky/skymodel.py:311:10: N806 variable 'Ionly_mask' in function should be lowercase
pyradiosky/skymodel.py:312:10: N806 variable 'NstokesI' in function should be lowercase
pyradiosky/skymodel.py:326:14: N806 variable 'rotation_matrix_T' in function should be lowercase
pyradiosky/skymodel.py:456:6: N806 variable 'Nside' in function should be lowercase
pyradiosky/skymodel.py:492:6: N806 variable 'flux_I' in function should be lowercase
pyradiosky/skymodel.py:494:10: N806 variable 'flux_I' in function should be lowercase
pyradiosky/skymodel.py:646:6: N806 variable 'Nsrcs' in function should be lowercase
pyradiosky/tests/test_utils.py:23:6: N806 variable 'stokesI' in function should be lowercase
pyradiosky/tests/test_utils.py:24:6: N806 variable 'stokesQ' in function should be lowercase
pyradiosky/tests/test_utils.py:25:6: N806 variable 'stokesU' in function should be lowercase
pyradiosky/tests/test_utils.py:26:6: N806 variable 'stokesV' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:176:6: N806 variable 'cosX' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:177:6: N806 variable 'sinX' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:203:6: N806 variable 'K_matrix' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:207:6: N806 variable 'I_matrix' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:290:6: N806 variable 'sinPsi' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:292:6: N806 variable 'cosPsi' in function should be lowercase
pyradiosky/spherical_coords_transforms.py:293:6: N806 variable 'Psi' in function should be lowercase
pyradiosky/tests/test_skymodel.py:167:6: N806 variable 'B' in function should be lowercase
pyradiosky/tests/test_skymodel.py:168:6: N806 variable 'J' in function should be lowercase
pyradiosky/tests/test_skymodel.py:232:6: N806 variable 'Jbeam' in function should be lowercase
pyradiosky/tests/test_skymodel.py:327:6: N806 variable 'Jbeam' in function should be lowercase
pyradiosky/tests/test_skymodel.py:357:6: N806 variable 'Nside' in function should be lowercase
pyradiosky/tests/test_skymodel.py:359:6: N806 variable 'Npix' in function should be lowercase
pyradiosky/tests/test_skymodel.py:473:6: N806 variable 'Nside' in function should be lowercase
pyradiosky/tests/test_skymodel.py:474:6: N806 variable 'Npix' in function should be lowercase
pyradiosky/tests/test_skymodel.py:582:6: N806 variable 'Nside' in function should be lowercase
pyradiosky/tests/test_skymodel.py:601:6: N806 variable 'Nside' in function should be lowercase
pyradiosky/tests/test_skymodel.py:602:6: N806 variable 'Npix' in function should be lowercase
pyradiosky/tests/test_skymodel.py:604:6: N806 variable 'Nfreqs' in function should be lowercase
pyradiosky/tests/test_skymodel.py:610:6: N806 variable 'Nskies' in function should be lowercase
pyradiosky/tests/test_skymodel.py:710:10: N806 variable 'sI' in function should be lowercase
pyradiosky/tests/test_skymodel.py:745:6: N806 variable 'Nsrcs' in function should be lowercase
pyradiosky/tests/test_skymodel.py:763:6: N806 variable 'minI_cut' in function should be lowercase
pyradiosky/tests/test_skymodel.py:764:6: N806 variable 'maxI_cut' in function should be lowercase
pyradiosky/tests/test_skymodel.py:781:6: N806 variable 'Ntimes' in function should be lowercase
pyradiosky/tests/test_skymodel.py:782:6: N806 variable 'Nsrcs' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:84:10: N806 variable 'R_screwy' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:85:10: N806 variable 'R_avg' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:88:10: N806 variable 'R_avg' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:104:10: N806 variable 'R_perturb' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:109:10: N806 variable 'R_perturb_pts' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:118:10: N806 variable 'R_exact' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:136:10: N806 variable 'Rv1' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:174:6: N806 variable 'sinPsi' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:176:6: N806 variable 'cosPsi' in function should be lowercase
pyradiosky/tests/test_spherical_coords_transforms.py:177:6: N806 variable 'Psi' in function should be lowercase
setup.py:19:8: N801 class name 'mock_version' should use CapWords convention
```

I don't know how rigorous we want to be with our pep8 naming or how akward it will be to change these names. I'm up for doing it, but will wait for other's opinions.

### CODE REVIEW

**Comment by aelanman** [comment]
I do _not_ recommend that we use the flake8-naming test, mainly because we have a standardization already for our attribute and variable naming in `pyuvdata` that other software that uses pyuvdata relies on. Enforcing a different convention here would make things more confusing. E.g., renaming `Ncomponents` to `n_components` breaks the pattern of `pyuvdata.UVData` attributes `Ntimes`, `Nblts`, `Nbls`, etc. Unless we're prepared to also overhaul `pyuvdata` and all the software depending on it, this seems like more trouble than it's worth.

In some cases, I worry that renaming things to match Python preferred convention would make things less clear. Capital `R` is a common notation for a rotation matrix, but lowercase `r` is not. `R_really_orthgonal` etc. are internal variables, it seems unnecessary to rename what are already descriptive names.

File: pyradiosky/skymodel.py [hunk 1]
```diff
@@ -55,15 +56,14 @@ class SkyModel(object):
...
-        Options :
-            - 'flat' : Flat spectrum.
-            - 'full' : Flux is defined by a saved value at each frequency.
-            - 'subband' : Flux is given at a set of band centers. (TODO)
-            - 'spectral_index' : Flux is given at a reference frequency. (TODO)
+        Options:
+        - 'flat' : Flat spectrum.
+        - 'full' : Flux is defined by a saved value at each frequency.
+        - 'subband' : Flux is given at a set of band centers. (TODO)
+        - 'spectral_index' : Flux is given at a reference frequency. (TODO)
```
Comment (bhazelton) [inline pyradiosky/skymodel.py]: We should check how this renders with Sphinx.

Comment (steven-murray) [reply pyradiosky/skymodel.py]: Yes we should... however we don't have a docs folder yet?

**Comment by steven-murray** [comment]
@aelanman Noted. I agree that sometimes mathematical variables are good to be capital. And if it was just one or two, I'd suggest doing `# noqa`, but that's gonna get ugly real fast. We could try turning off some of the pep8-naming rules, eg. N806? I think that would just leave one error, which should be easily fixable.

**Comment by bhazelton** [comment]
@steven-murray that seems like a reasonable compromise

**Pull request merged**